### PR TITLE
Feat/target nodepools

### LINKED ISSUE #1224: Feature: Allow loadbalancer to target specific nodes or nodepools
It would be great if the loadbalancer config also had a targetPools parameter. Claudie keeps the config up to date anyway, but instead of adding all nodes to the nginx config(or HA), only add the ones from the nodepools specified. 
That way i could still have nodes in different regions, and keep the traffic there, because of the loadbalancers only sending the traffic to specified nodes.

### PR DESCRIPTION
**Pull request by Despire**
Closes #1224 

`Target` is dropped in favor of `TargetPools` which specifies which nodepools used in the TargetK8s cluster of the LB cluster will be targeted by the Loadbalancer

e.g. in this config 
```
  kubernetes:
    clusters:
      - name: test-cluster
        version: v1.26.1
        network: 172.16.2.0/24
        pools:
          control:
            - hetzner
          compute:
            - hetzner-1
  loadBalancers:
    roles:
      - name: apiserver
        protocol: tcp
        port: 6443
        targetPort: 6443
        targetPools:
          - hetzner
      - name: dev-2
        protocol: tcp
        port: 6447
        targetPort: 6447
        targetPools:
          - hetzner-1
    clusters:
      - name: apiserver-lb-dev
        roles:
          - apiserver
        dns:
          dnsZone: claudie.dev
          provider: cloudflare-1
        targetedK8s: test-cluster
        pools:
          - hetzner
      - name: lb-dev-2
        roles:
          - dev-2
        dns:
          dnsZone: claudie.dev
          provider: cloudflare-1
        targetedK8s: test-cluster
        pools:
          - hetzner
```

LBcluster `apiserver-lb-dev` would target heztner pools in the K8s cluster `test-cluster`
LBcluster `lb-dev-2` would target hetzner-1 pool in the K8s cluster `test-cluster`

Using `Target` for new config changes is disabled, however there is still support for them to allow a transition period for clusters build with the old `Target` to the new `TargetPools`

The code for the old way will be removed in future releases

### CODE REVIEW

File: proto/config.proto [hunk 1]
```diff
@@ -127,8 +127,10 @@ message Role {
   int32 port = 3;
   // Port that load balancer will forward to.
   int32 targetPort = 4;
+  // TODO: remove in favor of targetPools.
```
[the remark this answers is not in the document]
Comment (Despire) [reply proto/config.proto:130]: Yes, for cluters build with the current version there will Target used in the MongoDB, i.e the current state will still have the `Target` field, this is just so we can still used the value from the current state when a workflow starts with this new version.

**Comment by Despire** [comment]
> For example user in the LB Target Pool will target a NodePool that is used for LB's?

This implementation only uses the Pools used in the K8s cluster. So even if the LBs use the same nodepool only the K8s "instances" will actually be targeted.

File: internal/manifest/manifest.go [hunk 2]
```diff
@@ -233,12 +233,8 @@ type Role struct {
 	Port int32 `validate:"min=0,max=65535" yaml:"port" json:"port"`
 	// Port where loadbalancer forwards the traffic.
 	TargetPort int32 `validate:"min=0,max=65535" yaml:"targetPort" json:"targetPort"`
-	// Defines a target group of nodes. Allowed values are:
-	// k8sAllNodes - All nodes in the cluster
-	// k8sControlNodes - Only control/master nodes in cluster
-	// k8sComputeNodes - Only compute/worker nodes in cluster
-	// +kubebuilder:validation:Enum=k8sAllNodes;k8sControlPlane;k8sComputePlane;
-	Target string `validate:"required,oneof=k8sAllNodes k8sControlPlane k8sComputePlane" yaml:"target" json:"target"`
```
[the remark this answers is not in the document]
Comment (bernardhalas) [reply internal/manifest/manifest.go:241]: I think we can drop this completely

**Comment by Despire** [comment]
There seems to be some issue when deleting nodes from the logs of the e2e pipeline will, delay this issue a bit longer

**Pull request merged**